replica/table.cc

```diff
@@ -993,6 +997,14 @@ tablet_storage_group_manager::maybe_split_sstable(const sstables::shared_sstable
     co_return co_await _t.get_compaction_manager().maybe_split_sstable(sst, cg.as_table_state(), split_compaction_options());
 }
 
+bool tablet_storage_group_manager::tombstone_gc_enabled(locator::tablet_id tid) const {
+    const locator::tablet_replica_set& replicas = tablet_map().get_tablet_info(tid).replicas;
+    const locator::tablet_replica this_replica { _my_host_id, this_shard_id() };
+    if (std::find(replicas.begin(), replicas.end(), this_replica) == replicas.end()) {
+        return false;
+    }
+    return true;
```

Comment:
nit: can just `return std::find(replicas.begin(), replicas.end(), this_replica) != replicas.end()`

Reply:
And how frequently is this function called?
We can precalculate this when the tablet_storage_group_manager is constructed and later in `tablet_storage_group_manager::update_effective_replication_map`.

Reply:
std::ranges::contains(replicas, this_replica)

Reply:
> And how frequently is this function called? We can precalculate this when the tablet_storage_group_manager is constructed and later in `tablet_storage_group_manager::update_effective_replication_map`.

Once for every partition (not row) that contains a tombstone. Compaction time is dominated by I/O and merging data, not iterating through a bounded replica set, so we thought it's not worth caching this info, since it's always better to have a single source of truth when possible.

Reply:
We can measure the impact with that perf sstable tool if you'd like to, against a sstable full of expiring tombstones.

Reply:
On a 2nd thought, maybe we don't want to waste cycles on this and the cache is worth the effort. If we go through the route of catching local tablet replicas transitioning from "pending" to "active" in update_erm(), the same mechanism can be reused for registering tables with views in the view update generator at the right time. Thoughts?

Reply:
How about small partitions?
Your assumptions could be skewed in this kind of dataset.

Reply:
I think it's worth caching the result in the storage group and updating it either incrementally and/or when the erm is updated, since those changes are relatively rare whild the per-partition check can be frequent with small partitions.

Reply:
> I think it's worth caching the result in the storage group and updating it either incrementally and/or when the erm is updated, since those changes are relatively rare whild the per-partition check can be frequent with small partitions.

I agree. @nodep please proceed with the change. It also has the benefit of reuse for fixing the view update from staging issue.